code/game/objects/structures/guillotine.dm

```diff
@@ -0,0 +1,234 @@
+
+					if (H)
+						H.regenerate_icons()
+
+					unbuckle_all_mobs()
+			else
+				blade_status = GUILLOTINE_BLADE_MOVING
+				icon_state = "guillotine_drop"
+				addtimer(CALLBACK(src, .proc/drop_blade, user), GUILLOTINE_ANIMATION_LENGTH)
+
+/obj/structure/guillotine/proc/raise_blade()
+	blade_status = GUILLOTINE_BLADE_RAISED
+	icon_state = "guillotine_raised"
+
+/obj/structure/guillotine/proc/drop_blade(mob/user)
+	if (buckled_mobs.len && blade_sharpness)
+		var/mob/living/carbon/human/H = buckled_mobs[1]
+
+		if (H)
+			var/obj/item/bodypart/head/head = H.get_bodypart("head")
+
+			if (head)
+				playsound(src, 'sound/weapons/bladeslice.ogg', 100, 1)
+				if (blade_sharpness >= GUILLOTINE_DECAP_MIN_SHARP || head.brute_dam >= 100)
+					head.dismember()
+					add_logs(user, H, "beheaded", src)
+					H.regenerate_icons()
+					unbuckle_all_mobs()
+					kill_count += 1
+
+					var/blood_overlay = "bloody"
+
+					if (kill_count == 2)
+						blood_overlay = "bloodier"
+					else if (kill_count > 2)
+						blood_overlay = "bloodiest"
+
+					blood_overlay = "guillotine_" + blood_overlay + "_overlay"
+					cut_overlays()
+					add_overlay(mutable_appearance(icon, blood_overlay))
+
+					// The crowd is pleased
+					// The delay is to making large crowds have a longer laster applause
+					var/delay_offset = 0
+					for(var/mob/M in viewers(src, 7))
+						var/mob/living/carbon/human/C = M
+						if (ishuman(M))
+							addtimer(CALLBACK(C, /mob/.proc/emote, "clap"), delay_offset * 0.3)
+							delay_offset++
+				else
+					H.apply_damage(15 * blade_sharpness, BRUTE, head)
+					add_logs(user, H, "dropped the blade on", src, " non-fatal")
+					H.emote("scream")
+
+				if (blade_sharpness > 1)
+					blade_sharpness -= 1
+
+	blade_status = GUILLOTINE_BLADE_DROPPED
+	icon_state = "guillotine"
+
+/obj/structure/guillotine/attackby(obj/item/W, mob/user, params)
+	if (istype(W, /obj/item/sharpener))
+		add_fingerprint(user)
+		if (blade_status == GUILLOTINE_BLADE_SHARPENING)
+			return
+		
+		if (blade_status == GUILLOTINE_BLADE_RAISED)
+			if (blade_sharpness < GUILLOTINE_BLADE_MAX_SHARP)
+				blade_status = GUILLOTINE_BLADE_SHARPENING
+				if(do_after(user, 7, target = src))
+					blade_status = GUILLOTINE_BLADE_RAISED
+					user.visible_message("<span class='notice'>[user] sharpens the large blade of the guillotine.</span>",
+						                 "<span class='notice'>You sharpen the large blade of the guillotine.</span>")
+					blade_sharpness += 1
+					playsound(src, 'sound/items/unsheath.ogg', 100, 1)
+					return
+				else
+					blade_status = GUILLOTINE_BLADE_RAISED
+					return
+			else
+				to_chat(user, "<span class='warning'>The blade is sharp enough!</span>")
+				return
+		else
+			to_chat(user, "<span class='warning'>You need to raise the blade in order to sharpen it!</span>")
+			return
+	else
+		return ..()
+
+/obj/structure/guillotine/buckle_mob(mob/living/M, force = FALSE, check_loc = TRUE)
```

Comment:
@ShizCalev I wonder if we should add the buckled mob's fingerprints here as well?